### PR TITLE
feat(js-sdk): run the vitest unit suite on Deno

### PR DESCRIPTION
**Pull request by mishushakov**
## Description

`pnpm test:deno` now runs the full vitest suite — the `unit` and `connectionConfig` projects, 421 sandbox/files/commands/pty/git/api/config tests — under the Deno runtime via `deno run -A npm:vitest run --project unit --project connectionConfig`, replacing the previous single dist-based smoke test (superseded — the suite covers the SDK under Deno far more thoroughly). The CI step runs on ubuntu only and covers the same projects as the Bun suite step from #1584, and the Deno pin is bumped from 1.46.3 to 2.8.1 (`setup-deno@v2`) since vitest needs Deno 2's Node compat.

Also drops the `edge` vitest project: `tests/runtimes/edge/` no longer exists, so it matched zero files.

Rebased on main after #1584: the off-Node fetch-caching fix originally in this PR was superseded by #1584's late-binding fix, which also makes the whole suite (including the per-proxy cache tests) pass under Deno with no test changes — so this PR is pure test/CI wiring.

Verified locally on Deno 2.8.1: unit project green (349 passed, 0 failed, 29 skipped — same skips as Node), connectionConfig project green (43 passed), and Node suite green.

## Usage

```bash
cd packages/js-sdk
pnpm test:deno
```

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by tvi — APPROVED** [review]

File: packages/js-sdk/package.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
     "pretest": "npx playwright install --with-deps chromium",
     "postPublish": "./scripts/post-publish.sh || true",
     "test:bun": "bunx --bun vitest run --project unit --project connectionConfig",
-    "test:deno": "deno test tests/runtimes/deno/ --allow-net --allow-read --allow-env --unstable-sloppy-imports --trace-leaks",
+    "test:deno": "deno run -A npm:vitest run --project unit",
```
[the remark this answers is not in the document]
Comment (mishushakov) [reply packages/js-sdk/package.json]: it's pinned in .tool-versions

[the remark this answers is not in the document]
Comment (mishushakov) [reply packages/js-sdk/package.json]: nothing got updated
